File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int ans = 0;
        for(int i=0;i<pow(2,nums.size());i++){
            auto c = i;
            int j=0;
            int temp = 0;
            while(c>0){
                int toTake = c&1;
                c>>=1;
                temp^=nums[j]*toTake;
                j++;
            }
            ans+=temp;
        }
        return ans;
    }
};

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        ans = 0
        for i in range(1,len(nums)+1):
            
            for comb in combinations(nums,i):
                ans += reduce(lambda a,b : a^b,comb)
        return ans
        # return sum(map(lambda x:reduce(lambda a,b: a^b,0),))